hedera-node/docs/design/services/token-service/airdrops/token-reject.md

```diff
@@ -0,0 +1,135 @@
+# Introduce TokenReject operation
+
+## Purpose
+
+As part of [HIP-904](https://hips.hedera.com/hip/hip-904), we would like to have the option of rejecting a token that has become part of an account's balance (no matter how e.g. via regular crypto transfer or airdrop).
+
+This would mean transferring the token (the whole amount if fungible) or the NFT (concrete serial number from a unique token type) back to its treasury account, without assessing custom fees. Using token reject an account could get rid of potential “ransom” tokens which have very expensive custom fees and only pay for the cheaper token reject operation.
```

Comment:
In the current code even a `CryptoTransfer` to treasury doesn't charge custom fees. 
So it seems like, the added advantage for this transaction is we can transfer back even the token is paused or token is frozen (which is not the case with `CryptoTransfer`)

Reply:
We don't have custom fees if the payer is the treasury account. 
However, it doesn't seem like we have a case where there is a fee exemption if the payer is sending to the treasury account.

Reference [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/customfees/CustomFeeExemptions.java#L51)

Reply:
@Neeharika-Sompalli Given the code Valio linked above it seems like we don't charge the custom fees only if the treasury is the sender of the transfer, but for transfers in the other direction (i.e. treasury is receiver) it looks like we charge custom fees from the sender and that's the case we can use `TokenReject` to avoid those fees.

Reply:
@MiroslavGatsanoga @MrValioBg yes I mean if the sender is treasury custom fees are[ not assessed.](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/CustomFeeAssessmentStep.java#L377). And if receiver is treasury royalty fees are not assessed, because thats the only fee receiver will be paying in custom fees [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/customfees/CustomRoyaltyFeeAssessor.java#L98).